repos/system_upgrade/el7toel8/actors/checkfstabxfsoptions/libraries/checkfstabxfsoptions.py

```diff
@@ -0,0 +1,58 @@
+from leapp import reporting
+from leapp.exceptions import StopActorExecutionError
+from leapp.libraries.stdlib import api
+from leapp.models import StorageInfo
+
+# man 5 xfs
+REMOVED_XFS_OPTIONS = set([
+    # removed from kernel in 4.0
+    'nodelaylog',
+    'delaylog',
+    'ihashsize',
+    'irixsgid',
+    'osyncisdsync',
+    'osyncisosync',
+    # removed from kernel in 4.19
+    'nobarrier',
+    'barrier',
+])
+
+
+def _get_storage_data():
+    storage = next(api.consume(StorageInfo), None)
+    if not storage:
+        raise StopActorExecutionError('The StorageInfo message is not available.')
+    if not storage.fstab:
+        raise StopActorExecutionError('Data from the /etc/fstab file is missing.')
+    return storage
+
+
+def process():
+    storage = _get_storage_data()
+    used_removed_options = set()
+    for entry in storage.fstab:
+        if entry.fs_vfstype == 'xfs':
+            options = set(entry.fs_mntops.split(','))
+            used_removed_options.update(options.intersection(REMOVED_XFS_OPTIONS))
+    used_removed_options = list(used_removed_options)
```

Comment:
```suggestion
```
Not necessary

Reply:
You are right. For a reason, I though that it's not possible to use set for the concatenation / join. fixed